### DISCUSSION TITLE
Need help with gulpInst[runMethod] is not a function

### DISCUSSION THREAD

**Discussion by anurag-jain** [gulpjs/gulp#2488] [Help]
Trying to compile CSS which are part of a [jekyll theme](https://github.com/jekyller/jasper2) and i did `npm install` followed by `gulp` and I run into following error.

`
aj@aj-LT:~/xyz.github.io$ gulp
[02:49:57] Using gulpfile ~/xyz.github.io/gulpfile.js
[02:49:57] gulpInst[runMethod] is not a function
[02:49:57] To list available tasks, try running: gulp --tasks
`
Here is output of `gulp -- version`
`
CLI version: 2.2.0
Local version: 3.9.1
`

**Comment by hovi** [comment]
I had this problem when I installed gulp using snap (versions):
```
CLI version: 2.2.0
Local version: 3.9.1
```
 When I installed it using npm, it worked:
```
CLI version: 2.3.0
Local version: 3.9.1
```

**Reply by phated** [reply]
Sounds like someone doesn't keep the snap up to date. That's not something we manage or control.

**Reply by EndaMolloy** [reply]
I'm also getting the same error message. @hovi was it solved simply by upgrading the CLI version to 2.3.0? I've also tried to do this but I'm struggling to remove 2.2.0...

**Reply by hovi** [reply]
I think so, you can just try and call directly gulp from local installation like this:
`node ./node_modules/gulp/bin/gulp.js`

**Reply by begin-again** [reply]
or `./node_modules/.bin/gulp`

**Reply by EndaMolloy** [reply]
> I think so, you can just try and call directly gulp from local installation like this:
> `node ./node_modules/gulp/bin/gulp.js`

That sorted it, thank you!

**Comment by sean-breeden** [comment]
I know this is an older post but I had the same problem in Ubuntu 20.04 just in case anyone is searching and needs this answer. Here is how I resolved it:

1. Uninstall gulp if you used apt to install it.
sudo apt remove gulp

2. Install npm
sudo apt install npm

3. Install nodejs
curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -
sudo apt install nodejs

4. Install gulp with npm
npm install gulp

5. sudo install gulp-cli
sudo npm install -g gulp-cli

6. Create symbolic link to gulp
sudo ln -s /usr/local/bin/gulp /usr/bin

**Reply by abobakr3** [reply]
Thank you , that's solved my issue .
